src/internet_identity/src/delegation.rs

```diff
@@ -45,15 +45,51 @@ pub async fn prepare_delegation(
     });
     update_root_hash();
 
-    state::usage_metrics_mut(|metrics| {
-        metrics.delegation_counter += 1;
-    });
+    delegation_bookkeeping(frontend);
+
     (
         ByteBuf::from(der_encode_canister_sig_key(seed.to_vec())),
         expiration,
     )
 }
 
+/// Update metrics and the list of latest front-end origins.
+fn delegation_bookkeeping(frontend: FrontendHostname) {
+    state::usage_metrics_mut(|metrics| {
+        metrics.delegation_counter += 1;
+    });
+    update_latest_delegation_origins(frontend);
+}
+
+/// Add the current front-end to the list of latest used front-end origins.
+fn update_latest_delegation_origins(frontend: FrontendHostname) {
+    let now = time();
```

Comment:
nit: maybe rename to `now_ns`, so that when it is used later (e.g. line 80 below), it is clear it is in nanoseconds?  (`timestamp` is somewhat more clear by convention, although using `timestamp_ns` could be also considered)